bundler/spec/runtime/with_unbundled_env_spec.rb

```diff
@@ -84,15 +84,16 @@ def run_bundler_script(env, script)
       expect(last_command.stdboth).to include "false"
     end
 
-    it "should remove '-rbundler/setup' from RUBYOPT" do
+    it "should remove '-rbundler/setup' from RUBYOPT even if it was present in original env" do
       create_file("source.rb", <<-RUBY)
         print #{modified_env}['RUBYOPT']
       RUBY
-      ENV["RUBYOPT"] = "-W2 -rbundler/setup #{ENV["RUBYOPT"]}"
+      setup_require = "-r#{lib_dir}/bundler/setup"
+      ENV["BUNDLER_ORIG_RUBYOPT"] = "-W2 #{setup_require} #{ENV["RUBYOPT"]}"
```

Comment:
Because we are trying to simulate that the RUBYOPT contained this `bundler/setup` value IN the original env, we actually have to set it in `BUNDLER_ORIG_RUBYOPT` for the test. We are simulating that this `bundler/setup` entry in RUBYOPT existed BEFORE bundler was activated (even though it didn't).

Otherwise, if we set it in `RUBYOPT` it will already be removed even without the code in `unbundled_env` (The way the spec was written before it still passed when I commented out the code in `unbundled_env` for handling RUBYOPT).

That is because `unbundled_env` calls `original_env` as it's starting point, which will remove whatever we set here in RUBYOPT because it was not in the original env when the test suite first activated bundler.